cupertino-adaptive/src/commonMain/kotlin/io/github/alexzhirkevich/cupertino/adaptive/AdaptiveTopAppBar.kt

```diff
@@ -67,23 +68,58 @@ fun AdaptiveTopAppBar(
             )
         },
         material = {
-            TopAppBar(
+            SingleRowTopAppBar(
                 title = title,
+                centeredTitle = it.centeredTitle,
+                colors = it.colors,
                 modifier = modifier,
                 navigationIcon = navigationIcon,
                 actions = actions,
                 windowInsets = windowInsets,
-                colors = it.colors
             )
         }
     )
 }
 
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun SingleRowTopAppBar(
+    title: @Composable () -> Unit,
+    centeredTitle: Boolean,
+    colors: TopAppBarColors,
+    modifier: Modifier = Modifier,
+    navigationIcon: @Composable () -> Unit = {},
+    actions: @Composable (RowScope.() -> Unit) = {},
+    windowInsets: WindowInsets = CupertinoTopAppBarDefaults.windowInsets,
+) {
+    if (centeredTitle) {
+        CenterAlignedTopAppBar(
+            title = title,
+            modifier = modifier,
+            navigationIcon = navigationIcon,
+            actions = actions,
+            windowInsets = windowInsets,
+            colors = colors,
+        )
+    } else {
+        TopAppBar(
+            title = title,
+            modifier = modifier,
+            navigationIcon = navigationIcon,
+            actions = actions,
+            windowInsets = windowInsets,
+            colors = colors,
+        )
+    }
+}
+
 @Stable
 @OptIn(ExperimentalMaterial3Api::class)
 class MaterialTopAppBarAdaptation internal constructor(
-    colors : TopAppBarColors
+    colors: TopAppBarColors,
+    centeredTitle: Boolean = false,
 ) {
+    var centeredTitle: Boolean by mutableStateOf(centeredTitle)
```

Comment:
It's better to use `isCenterAligned` instead

Reply:
Fixed